pkg/registry/apis/folders/legacy_storage.go

```diff
@@ -0,0 +1,265 @@
+	if err != nil {
+		return nil, err
+	}
+
+	dto, err := s.service.Get(ctx, &folder.GetFolderQuery{
+		SignedInUser: user,
+		UID:          &name,
+		OrgID:        info.OrgID,
+	})
+	if err != nil || dto == nil {
+		if errors.Is(err, dashboards.ErrFolderNotFound) || err == nil {
+			err = resourceInfo.NewNotFound(name)
+		}
+		return nil, err
+	}
+
+	return convertToK8sResource(dto, s.namespacer), nil
+}
+
+func (s *legacyStorage) Create(ctx context.Context,
+	obj runtime.Object,
+	createValidation rest.ValidateObjectFunc,
+	options *metav1.CreateOptions,
+) (runtime.Object, error) {
+	info, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	p, ok := obj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, fmt.Errorf("expected folder?")
+	}
+
+	// Simplify creating unique folder names with
+	if p.GenerateName != "" && strings.Contains(p.Spec.Title, "${RAND}") {
+		rand, _ := util.GetRandomString(10)
+		p.Spec.Title = strings.ReplaceAll(p.Spec.Title, "${RAND}", rand)
+	}
+
+	accessor := kinds.MetaAccessor(p)
+	parent := accessor.GetFolder()
+
+	out, err := s.service.Create(ctx, &folder.CreateFolderCommand{
+		SignedInUser: user,
+		UID:          p.Name,
+		Title:        p.Spec.Title,
+		Description:  p.Spec.Description,
+		OrgID:        info.OrgID,
+		ParentUID:    parent,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return s.Get(ctx, out.UID, nil)
+}
+
+func (s *legacyStorage) Update(ctx context.Context,
+	name string,
+	objInfo rest.UpdatedObjectInfo,
+	createValidation rest.ValidateObjectFunc,
+	updateValidation rest.ValidateObjectUpdateFunc,
+	forceAllowCreate bool,
+	options *metav1.UpdateOptions,
+) (runtime.Object, bool, error) {
+	info, err := request.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, false, err
+	}
+
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	created := false
+	oldObj, err := s.Get(ctx, name, nil)
+	if err != nil {
+		return oldObj, created, err
+	}
+
+	obj, err := objInfo.UpdatedObject(ctx, oldObj)
+	if err != nil {
+		return oldObj, created, err
+	}
+	f, ok := obj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, created, fmt.Errorf("expected folder after update")
+	}
+	old, ok := oldObj.(*v0alpha1.Folder)
+	if !ok {
+		return nil, created, fmt.Errorf("expected old object to be a folder also")
+	}
+
+	oldParent := kinds.MetaAccessor(old).GetFolder()
+	newParent := kinds.MetaAccessor(f).GetFolder()
+	if oldParent != newParent {
+		_, err = s.service.Move(ctx, &folder.MoveFolderCommand{
+			SignedInUser: user,
+			UID:          name,
+			OrgID:        info.OrgID,
+			NewParentUID: newParent,
+		})
+		if err != nil {
+			return nil, created, fmt.Errorf("error changing parent folder spec")
+		}
+	}
+
+	changed := false
+	cmd := &folder.UpdateFolderCommand{
+		SignedInUser: user,
+		UID:          name,
+		OrgID:        info.OrgID,
```

Comment:
```suggestion
		OrgID:        info.OrgID,
		Overwrite:    true,
```
otherwise, it fails with:
```
for: "pkg/tests/apis/folders/testdata/folder-generate.yaml": error when patching "pkg/tests/apis/folders/testdata/folder-generate.yaml": the folder has been changed by someone else
```
In a follow up we could conditionally set it but I think it's ok to set it to true for now.

I also get the following warning:
```
Warning: resource folders/ccb7f960-a426-4b9a-a91f-04da6d7df12a is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
```

Reply:
overwrite=true should be fine, we won't even get to this unless the revisionVersion is valid

The `kubectl.kubernetes.io/last-applied-configuration` warning is in the frontend -- because we are not saving their annotation.  The fallback behavior is to do server-side-apply, and that seems to work OK